### PR TITLE
Move compatibility section to new documentation structure

### PR DESCRIPTION
**Pull request by alfrunes**
Help wanted: I could not resolve all internal references with the new documentation structure and commented them out with a FIXME tag.

### CODE REVIEW

**Comment by lluiscampos** [comment]
> Help wanted: I could not resolve all internal references with the new documentation structure and commented them out with a FIXME tag.

During my work I am also linking to non existing pages. I think it is okey to have dead links as we will run a full link check once we are getting closer.

**Comment by alfrunes** [comment]
> > Help wanted: I could not resolve all internal references with the new documentation structure and commented them out with a FIXME tag.
> 
> During my work I am also linking to non existing pages. I think it is okey to have dead links as we will run a full link check once we are getting closer.

OK, I will just insert some tentative links then.
I also have some passive voice corrections coming up as well.

**Comment by alfrunes** [comment]
@kacf or @lluiscampos; I don't understand why the autoversion check fails at that particular section - it is literally copy-paste from the old docs.

File: 02.Overview/10.Compatibility/docs.md [hunk 1]
```diff
@@ -4,3 +4,126 @@ taxonomy:
...
+
+!!! <sup>1</sup> For early versions of Yocto, consult the documentation for that specific Mender version using the left hand menu.
+
+!! <sup>2</sup> Rolling back to 1.x.x from a failed upgrade to 2.x.x is supported. However, it is not possible to downgrade to a Mender 1.x.x client from a 2.x.x client, once the update containing 2.x.x has been committed.
+
+<!--AUTOVERSION: "Yocto % branch and earlier"/ignore-->
+!!! <sup>3</sup> For compatibility reasons, the Yocto thud branch and earlier use Mender 1.7 by default. Please see [the section on configuring the build](../../system-updates/yocto-project/customize-mender#configuring-the-build) for how to force a later version.
+
+<!--AUTOVERSION: "from % to %"/ignore "from-%-to-%"/ignore-->
+! <sup>4</sup> If upgrading from thud to warrior, see also [known issues when upgrading from thud to warrior](../../troubleshoot/yocto-project-build#upgrading-to-warrior-from-thud)
```
Comment (kacf) [inline 02.Overview/10.Compatibility/docs.md:83]: > @kacf or @lluiscampos; I don't understand why the autoversion check fails at that particular section - it is literally copy-paste from the old docs.

It's not exactly copy-paste. It has gone from 

* `from-thud-to-warrior`

to

* `to-warrior-from-thud`

so the AUTOVERSION expression needs to follow as well.

Comment (alfrunes) [reply 02.Overview/10.Compatibility/docs.md:83]: Ohh, I forgot about the reference/link I rewrote. Thanks!

**Comment by lluiscampos** [comment]
> LGTM, but should the old chapter be removed?

The old chapters don't exist in `documentation-restructuring` branch, we are adding them from master.

File: 02.Overview/10.Compatibility/docs.md [hunk 2]
```diff
@@ -4,3 +4,126 @@ taxonomy:
...
     label: reference
 ---
+
+The following section outlines the compatibility between versions of Mender components, such as the client and server.
+
+
+## Backward compatibility policy
+
+<!--AUTOVERSION: "% to %"/ignore-->
+Mender always provide an [upgrade path](../../server-installation/upgrade) from the past patch (e.g. 1.2.0 to 1.2.1) and minor version (e.g. 1.1.1 to 1.2.0), and releases follow [Semantic Versioning](http://semver.org/?target=_blank). Note that Semantic Versioning allows new functionality to be added between minor releases so be sure to upgrade components to support the newer functionality before starting to use it.
```
Comment (drewmoseley) [inline 02.Overview/10.Compatibility/docs.md:14]: always provides

Comment (drewmoseley) [reply 02.Overview/10.Compatibility/docs.md:14]: releases following [Semantic Versioning]

Comment (drewmoseley) [reply 02.Overview/10.Compatibility/docs.md:14]: I don't follow here what actually is allowed.  I think we should clarify.  Something like:
Mender releases follow Semantic versioning which uses 3 numeric fields such as 1.2.3 where 1 is the major release, 2 is the minor release and 3 is the patch release.  Releases within a major release number will retain compatibility.

File: 02.Overview/10.Compatibility/docs.md [hunk 3]
```diff
@@ -4,3 +4,126 @@ taxonomy:
...
+| Mender 2.3.x / mender-artifact 3.3.x | no          | yes         | yes         |
+
+!! Older Mender clients do not support newer versions of the Artifact format; they will abort the deployment. You can build older versions of the Mender Artifact format to upgrade older Mender clients. See [Write a new Artifact](../../artifacts/modifying-a-mender-artifact#create-an-artifact-from-a-raw-root-file-system) for an introduction how to do this.
+
+
+## Mender server and client API
+
+The Mender client uses the Devices API to communicate with the server. Documentation for the Devices API are available for both [Open Source](../../apis/open-source/device-apis) and [Mender Enterprise](../../apis/enterprise/device-apis). Even though semantic versioning applies to both the Mender client and server, we also emphasize that the server minor version puts an upper limit on compatible client versions within the same major version. Moreover, please ensure that the client and server support the [Artifact format](#mender-clientserver-and-artifact-format) version you are using. The following table shows the compatibility requirements between the Mender client with the API exposed by the server.
```
Comment (drewmoseley) [inline 02.Overview/10.Compatibility/docs.md:113]: we emphasize that the server... 
(ie the "also" is redundant here)

**Comment by kacf** [comment]
> @kacf The old chapters don't exist in `documentation-restructuring` branch, we are adding them from master.

Ah, it's starting with nothing, got it!

File: 02.Overview/10.Compatibility/docs.md [hunk 4]
```diff
@@ -4,3 +4,128 @@ taxonomy:
...
     label: reference
 ---
+
+The following section outlines the compatibility between versions of Mender components, such as the client and server.
+
+
+## Backward compatibility policy
+
+<!--AUTOVERSION: "e.g. %"/ignore-->
+Mender releases follow [Semantic Versioning](http://semver.org) which uses three numeric fields on the form MAJOR.MINOR.PATCH (e.g. 1.2.3) to describe a release. An increment in the *major* version signifies backward-incompatible changes with the previous release, where as incrementing the *minor* version indicates new features are added without breaking backward compatibility. For all other changes that does not change the functional behavior (i.e. bug fixes) results in incrementing *patch* version. Releases within a *major* number will retain compatibility for any *minor* and *patch* version<sup>1</sup>. Please have a look at the [section on upgrading](../../server-installation/upgrade) for details on upgrading Mender.
```
Comment (drewmoseley) [inline 02.Overview/10.Compatibility/docs.md:14]: fields of the form...
For all other changes that do not change the functional behavior (i.e. bug fixes), the *patch* version is incremented.

**Pull request merged**